synapse/handlers/typing.py

```diff
@@ -275,7 +275,7 @@ def _push_update_local(self, member, typing):
                 typing = self._room_typing[room_id]
                 rows.append((serial, room_id, list(typing)))
         rows.sort()
-        return rows
+        return rows[:limit]
```

Comment:
eyebrow raised: we're just dropping updates beyond the limit?

Reply:
The current situation is that the replication sees that it is limited and panics. The next ~~episode~~ PR in the series changes it so that if the stream is limited then it gets the token from the last item in the returned list

Reply:
(ftr: the relevant PR is #7024)

Reply:
conclusions from discussion elsewhere:

* the fact that we ignore updates between `limit` and `current_id` is not unique to this function: all of the `update_function`s do the same thing (it just looks a bit different here because the typing stream is in-memory)
* if we hit the limit, then we will throw an exception and disconnect the stream: https://github.com/matrix-org/synapse/blob/1410dab7642227531b4e30a135dccab506702fb6/synapse/replication/tcp/streams/_base.py#L200
* slicing the result here is mostly done for consistency with the other `update_functions`

Reply:
further discussion:

> * slicing the result here is mostly done for consistency with the other `update_functions`

... but also because we're going to rely on the `update_functions` honouring their `limit` in a future PR, so we should bring this one in line now even though it's not technically necessary yet.